app/src/main-process/squirrel-updater.ts

```diff
@@ -92,12 +98,20 @@ function writeBatchScriptCLITrampoline(binPath: string): Promise<void> {
   return writeFile(trampolinePath, trampoline)
 }
 
+/** We have to make sure the versioned path in the script has the correct format
+ * even if its Windows otherwise there can be edge cases with WSL see issue:
+ * https://github.com/desktop/desktop/issues/4998
+ */
 function writeShellScriptCLITrampoline(binPath: string): Promise<void> {
-  const versionedPath = resolveVersionedPath(
+  const rawVersionedPath = resolveVersionedPath(
     binPath,
     'resources/app/static/github.sh'
   )
 
+  const versionedPath = __WIN32__
+    ? rawVersionedPath.replace(/\\/g, '/')
+    : rawVersionedPath
+
```

Comment:
`writeShellScriptCLITrampoline` is only called when on Windows. macOS has a [separate method](https://github.com/desktop/desktop/blob/2a9d7d6d45c4fe1876008803f9e5c6d51d24eb14/app/src/ui/lib/install-cli.ts) for installing its CLI trampoline.

With that in mind I think we can simplify this

```suggestion
  const versionedPath = resolveVersionedPath(
    binPath,
    'resources/app/static/github.sh'
  ).replace(/\\/g, '/')
```